core/src/unprocessed_transaction_storage.rs

```diff
@@ -25,6 +23,9 @@ pub enum UnprocessedTransactionStorage {
     LocalTransactionStorage(ThreadLocalUnprocessedPackets),
 }
 
+unsafe impl Send for UnprocessedTransactionStorage {}
```

Comment:
I'm not convinced we can actually do this w/o introducing race conditions on the reference counts for votes.

Even though we wrap in an RwLock in the VoteStorage structure, I think it's possible another thread could drop or clone the Rc from another reference to it. Then because it is non-atomically updated, we end up with a miscounted Rc.

Reply:
Additionally, in my local changes for the tx-scheduler, I am *actually* sharing these references between threads, and converted the Rc to Arc.

It seems it'd be useful if I split that change out so we could use it here. I'll do that shortly.

Reply:
^ I split out #28145 if we go that route here

Reply:
I agree, this only works in this specific situation where there is only 1 thread dropping the Rc - even then it's pretty sketch. I would also prefer it to be an Arc. I can hold off and wait for #28145 .